File: front-end-ui/src/app/models/card.ts
export class Card {
    constructor(name: string, type: string, quantity: number) {
        this.name = name;
        this.type = type;
        this.quantity = quantity;
    }

    public name: string;
    public type: string;
    public quantity: number;
    public cardTypeId!: number;
}

File: Team4.Seshat.Model/Models/Card.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace Team4.Seshat.Model.Models
{
    [Table("CardRecord")]
    public class Card
    {
        public int Id { get; set; }
        public string Name { get; set; }
        [Column("Category")]
        public string Type { get; set; }
        public int? CardTypeId { get; set; }
        [ForeignKey("CardTypeId")]
        public virtual CardType CardType { get; set; }
        public int Quantity { get; set; }
        public int LibraryId { get; set; }
    }
}
